add-on/src/lib/ipfs-client/embedded-chromesockets/config.js

```diff
@@ -113,7 +124,36 @@ async function buildConfig (opts, log) {
   // merge configs
   const finalOpts = {
     start: false,
-    libp2p: chromeSocketsBundle
+    libp2p: ({ libp2pOptions, peerInfo }) => {
+      const wrtcstar = new WebRTCStar({ id: peerInfo.id })
+
+      // TODO: provisional reset in case the same code run already in js-ipfs/src/core/runtime/libp2p-browser.js
+      peerInfo.multiaddrs.replace('/p2p-websocket-star', chromeDefaultOpts.config.Addresses.Swarm.find(addr => addr.includes('p2p-websocket-star')))
+
+      // this can be replaced once optional listening is supported with the below code. ref: https://github.com/libp2p/interface-transport/issues/41
+      // const wsstar = new WebSocketStar({ id: _options.peerInfo.id, ignore_no_online: true })
+      const wsstarServers = peerInfo.multiaddrs.toArray().map(String).filter(addr => addr.includes('p2p-websocket-star') && !addr.startsWith('/p2p-websocket-star'))
+      log('ws-star servers', wsstarServers)
+      peerInfo.multiaddrs.replace(wsstarServers.map(multiaddr), '/p2p-websocket-star') // the ws-star-multi module will replace this with the chosen ws-star servers
+      const wsstar = new WebSocketStarMulti({ servers: wsstarServers, id: peerInfo.id, ignore_no_online: true }) // allow scenario when all ws-stars are offline
+
+      libp2pOptions.modules.transport = [
+        TCP,
+        WS,
+        wrtcstar,
+        wsstar
+      ]
+
+      libp2pOptions.modules.peerDiscovery = [
+        wrtcstar.discovery,
+        wsstar.discovery,
+        MulticastDNS,
+        Bootstrap
+      ]
+
+      log('initializing libp2p with libp2pOptions', libp2pOptions)
+      return new Libp2p(libp2pOptions)
```

Comment:
If we ignore necessary ws-star weirdness and inline code, this is basically 5 lines instead of 200  :+1: